galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/TestClassesRas.java

```diff
@@ -0,0 +1,76 @@
+
+package dev.galasa.framework.api.ras.internal;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import dev.galasa.framework.spi.IFramework;
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+    "osgi.http.whiteboard.servlet.pattern=/ras/testclasses" }, name = "TestClasses RAS")
+
+public class TestClassesRas extends HttpServlet{
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
+    
+    private Gson gson = new Gson();
+    private final List<String> DUMMYQUERY = Arrays.asList("requestor", "from", "to", "testclasses", "page", "size");
+    private final List<String> TESTCLASSDUMMY = Arrays.asList("testclass","xxxxxxx","bundle","yyyyyy","testclass","xxxxxxx","bundle","yyyyyy");
+
+    @Reference
+    public IFramework framework; // NOSONAR
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        boolean found = false;
+        JsonObject testclasses = new JsonObject();
+        JsonArray testObjects = new JsonArray();
+        try{ 
+            for(String search: DUMMYQUERY){
+                if(search=="testclasses"){
```

Comment:
Sorry Diana,  this is a real java rockie mistake.  You cannot compare Strings in this way.   What you are asking for is, is the java object the same,  not are the Strings the same, therefore this will always fail.
if (search.equals("testclasses")) is the correct way of comparing Strings.

Reply:
Second point,   why are you searching for testclasses in the query?  the fact the servlet was called, means testclasses are needed.  no need to search the query path for it.   All you need to search the query path for is the sort options